### PR TITLE
Rebase/element android 1.4.11

### PR DESCRIPTION
**Pull request by Claire1817**
#541

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomViewModel.kt [hunk 1]
```diff
@@ -146,24 +155,7 @@ class CreateDirectRoomViewModel @AssistedInject constructor(
...
-            val roomParams = CreateRoomParams()
-                    .apply {
-                        selections.forEach {
-                            when (it) {
-                                is PendingSelection.UserPendingSelection     -> invitedUserIds.add(it.user.userId)
-                                is PendingSelection.ThreePidPendingSelection -> invite3pids.add(it.threePid)
-                            }
-                        }
-                        setDirectMessage()
-                        enableEncryptionIfInvitedUsersSupportIt = adminE2EByDefault
```
Comment (yostyle) [inline vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomViewModel.kt]: What is new among this part in EA ?

File: vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomViewModel.kt [hunk 2]
```diff
@@ -197,9 +188,6 @@ class CreateDirectRoomViewModel @AssistedInject constructor(
             }
         }
     }
-=======
-            analyticsTracker.capture(CreatedRoom(isDM = roomParams.isDirect.orFalse()))
```
Comment (yostyle) [inline vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomViewModel.kt]: Maybe add this change and comment it ?

File: vector/src/main/java/im/vector/app/features/home/room/detail/TimelineViewModel.kt [hunk 3]
```diff
@@ -719,22 +719,12 @@ class TimelineViewModel @AssistedInject constructor(
...
-<<<<<<< HEAD
-                R.id.open_matrix_apps          -> false // Tchap: there are no matrix apps
-                R.id.voice_call                -> BuildConfig.IS_VOIP_SUPPORTED && state.isWebRTCCallOptionAvailable() // Tchap: check if voip is enabled
-                R.id.video_call                -> BuildConfig.IS_VOIP_SUPPORTED && // Tchap: check if voip is enabled
-                        (state.isWebRTCCallOptionAvailable() || state.jitsiState.confId == null || state.jitsiState.hasJoined)
-                // Show Join conference button only if there is an active conf id not joined. Otherwise fallback to default video disabled. ^
-                R.id.join_conference           -> !state.isWebRTCCallOptionAvailable() && state.jitsiState.confId != null && !state.jitsiState.hasJoined
-                R.id.search                    -> true
-=======
                 R.id.open_matrix_apps          -> true
```
Comment (yostyle) [inline vector/src/main/java/im/vector/app/features/home/room/detail/TimelineViewModel.kt:722]: Tchap is not recognized as a matrix app. Please restore the change coming from Tchap.

File: vector/src/main/java/im/vector/app/features/home/room/detail/TimelineViewModel.kt [hunk 4]
```diff
@@ -719,22 +719,12 @@ class TimelineViewModel @AssistedInject constructor(
...
                 R.id.voice_call                -> state.isCallOptionAvailable()
                 R.id.video_call                -> state.isCallOptionAvailable() || state.jitsiState.confId == null || state.jitsiState.hasJoined
```
Comment (yostyle) [inline vector/src/main/java/im/vector/app/features/home/room/detail/TimelineViewModel.kt:724]: Please check if voip is enabled.

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/MessageItemFactory.kt [hunk 5]
```diff
@@ -161,6 +134,7 @@ class MessageItemFactory @Inject constructor(
     private val messageItemAttributesFactory: MessageItemAttributesFactory,
     private val contentUploadStateTrackerBinder: ContentUploadStateTrackerBinder,
     private val contentDownloadStateTrackerBinder: ContentDownloadStateTrackerBinder,
+    private val contentScannerStateTracker: ContentScannerStateTracker,
```
Comment (yostyle) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/MessageItemFactory.kt]: Did you test content scanner feature ?

File: vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthResetPasswordFragment.kt [hunk 6]
```diff
@@ -120,24 +115,10 @@ class FtueAuthResetPasswordFragment @Inject constructor() : AbstractFtueAuthFrag
...
-            is Fail    -> {
-                views.resetPasswordEmailTil.error = errorFormatter.toHumanReadable(state.asyncResetPassword.error)
-            }
-            is Success -> Unit
-=======
         setupUi(state)
         if (state.isLoading) {
             // Ensure new password is hidden
             views.passwordField.hidePassword()
->>>>>>> v1.4.11
```
Comment (yostyle) [inline vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthResetPasswordFragment.kt]: I don't understand why we have conflicts in this file. it seems we don't use it in Tchap, right ?

File: .github/workflows/triage-move-review-requests.yml [hunk 7]
```diff
@@ -7,6 +7,8 @@ jobs:
   add_design_pr_to_project:
     name: Move PRs asking for design review to the design board
     runs-on: ubuntu-latest
+    # Skip in forks
+    if: github.repository == 'vector-im/element-android'
```
[the remark this answers is not in the document]
Comment (Claire1817) [reply .github/workflows/triage-move-review-requests.yml:11]: yes

File: vector/build.gradle [hunk 8]
```diff
@@ -581,17 +577,10 @@ dependencies {
     // WebRTC
     // org.webrtc:google-webrtc is for development purposes only
     // implementation 'org.webrtc:google-webrtc:1.0.+'
-<<<<<<< HEAD
-    withvoipImplementation('com.facebook.react:react-native-webrtc:1.92.1-jitsi-9093212@aar')
-
-    // Jitsi
-    withvoipImplementation('org.jitsi.react:jitsi-meet-sdk:3.10.0') {
```
[the remark this answers is not in the document]
Comment (Florian14) [reply vector/build.gradle:584]: I confirm, we should keep the lib version from Element and use `withvoipImplementation`

[the remark this answers is not in the document]
Comment (Claire1817) [reply vector/build.gradle:584]: So, i keep withvoipImplementation but i update the version to 5.0.2 ? @Florian14

[the remark this answers is not in the document]
Comment (yostyle) [reply vector/build.gradle:584]: > So, i keep withvoipImplementation but i update the version to 5.0.2 ? @Florian14

You must use the same version as EA.

File: vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomViewModel.kt (hunk 1 above)
Comment (Florian14) [reply vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomViewModel.kt]: We have modified a lot this class in Tchap, I probably made the choice to remove the unused code for clarity

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/MessageItemFactory.kt (hunk 5 above)
Comment (Claire1817) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/MessageItemFactory.kt]: yes and it's ok

File: vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomViewModel.kt (hunk 1 above)
Comment (yostyle) [reply vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomViewModel.kt]: @Claire1817 could you take a look on file history to verify that's florian said.

File: vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomViewModel.kt [hunk 9]
```diff
@@ -293,47 +291,47 @@ class CreateRoomViewModel @AssistedInject constructor(
...
-                    }
+//                    // Encryption
+//                    val shouldEncrypt = when (state.roomJoinRules) {
+//                        // we ignore the isEncrypted for public room as the switch is hidden in this case
+//                        RoomJoinRules.PUBLIC -> false
+//                        else                 -> state.isEncrypted ?: state.defaultEncrypted[state.roomJoinRules].orFalse()
+//                    }
+//                    if (shouldEncrypt) {
+//                        enableEncryption()
+//                    }
```
[the remark this answers is not in the document]
Comment (yostyle) [reply vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomViewModel.kt:334]: Ok you could remove this code as florian already did that in the past. Please a Tchap comment.

**Review by yostyle — APPROVED** [review]

**Review by Florian14 — APPROVED** [review]

**Pull request merged**